ethsigner/signingapi/src/main/java/tech/pegasys/ethsigner/core/signing/TransactionSignerFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.signing;
+
+import java.lang.reflect.InvocationTargetException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class TransactionSignerFactory {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  public static TransactionSigner create(final TransactionSignerConfig config) {
```

Comment:
I'd avoid plugins ... and make this super simple ....

Reply:
OK, I could just match a String to an implementation (Map)

Reply:
Could you make use of method overloading, varying the input class type?

Reply:
I've made it a very simple if/else thing now using new, because nobody seemed to like my reflection bit ... :-(